purchase_fop_shipping/models/purchase.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# © 2017 Akretion (http://www.akretion.com)
+#   @author Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.exceptions import Warning as UserError
+from openerp import models, fields, api, _
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+    fop_reached = fields.Boolean(
+        string='FOP reached',
+        help='Free-Of-Paiment shipping reached',
+        compute='_fop_shipping_reached')
+    force_order_under_fop = fields.Boolean(
+        string='Confirm under FOP',
+        help='Force confirm order under Free-Of-Paiment shipping',)
+    fop_shipping = fields.Float(
```

Comment:
I think readonly is more safe. In some use case supplier can accept to apply exceptionaly a low a FOP for 1 order. Some user can think that the change is only for the current purchase, but it s not true.